sympy/functions/elementary/trigonometric.py

```diff
@@ -117,10 +117,19 @@ def _peeloff_pi(arg):
             K, p = a.as_two_terms()
             if p is S.Pi and K.is_Rational:
                 break
+            K = a.coeff(S.Pi)
```

Comment:
Why k is here as the coeff of pi? Can you please explain the logic applied here so that if someone traceback then he/she could understand the logic

Reply:
If we get a pi term and its coeff is rational, we can use it to express it in the form which is required for the peeloff function, i.e Split ARG into two parts, a "rest" and a multiple of pi/2.